CHANGELOG.md

```diff
@@ -5,6 +5,7 @@
 ### Features
 
 ### Bug fixes
+* Changed `BiologicalSequence.distance` to raise an error any time two sequences are passed of different lengths regardless of the `distance_fn` being passed. Fixes #514
```

Comment:
The issue number should link to the issue -- take a look at the other changelog messages for examples of how to format this. I also recommend putting the issue number in parentheses for consistency.